archive_updater_test.py

```diff
@@ -0,0 +1,65 @@
+import unittest
+import subprocess
+import os
+import tempfile
+import shutil
+from archive_updater import fill_message_directory
+from unittest import mock
+
+class ArchiveUpdaterFillMessageDirectoryTest(unittest.TestCase):
+
+    def setUp(self):
+        self.tmp_dir = tempfile.mkdtemp()
+    def tearDown(self):
+        shutil.rmtree(self.tmp_dir)
+
+    @mock.patch.object(subprocess, 'check_call')
+    def test_update_failure(self, mock_check_call):
+        mock_check_call.side_effect = subprocess.CalledProcessError('Failed to fetch', 'git fetch')
+        self.assertRaises(subprocess.CalledProcessError,
+                          fill_message_directory, '../linux-kselftest/git/0.git',
+                                                 '../lkml-gerrit-bridge/test_data',
+                                                 '5227e860278407173cbfdab594676764153c27e1')
+
+    @mock.patch.object(subprocess, 'check_output')
+    @mock.patch.object(subprocess, 'call')
+    @mock.patch.object(subprocess, 'check_call')
+    def test_fill_message_directory_success(self, mock_check_call, mock_call, mock_check_output):
+        def writeFile(args, stdout):
+            stdout.write('called with: ' + ' '.join(args))
+        mock_check_call.return_value = ''
```

Comment:
`mock_check_call` shouldn't return anything.
Drop this line.

This also keeps the def of `writeFile` closer to where it's used (the line below this)